File: benchmarks/pandas/bench_multi_index_set_names.py
"""Benchmark: MultiIndex.set_names() on 100k-pair MultiIndex"""
import json, time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10
a = [f"a{i % 100}" for i in range(ROWS)]
b = [i % 1000 for i in range(ROWS)]
tuples = list(zip(a, b))
mi = pd.MultiIndex.from_tuples(tuples)

for _ in range(WARMUP):
    mi.set_names(["level0", "level1"])

start = time.perf_counter()
for _ in range(ITERATIONS):
    mi.set_names(["level0", "level1"])
total = (time.perf_counter() - start) * 1000
print(json.dumps({"function": "multi_index_set_names", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_multi_index_set_names.ts
/**
 * Benchmark: MultiIndex.setNames() on 100k-pair MultiIndex
 */
import { MultiIndex } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;
const a = Array.from({ length: ROWS }, (_, i) => `a${i % 100}`);
const b = Array.from({ length: ROWS }, (_, i) => i % 1000);
const tuples: [string, number][] = a.map((v, i) => [v, b[i] as number]);
const mi = new MultiIndex({ tuples });

for (let i = 0; i < WARMUP; i++) mi.setNames(["level0", "level1"]);
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) mi.setNames(["level0", "level1"]);
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "multi_index_set_names",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
